In [ ]:
import pandas as pd
import numpy as np

In [ ]:
df = pd.read_csv('telco_customer_churn.csv')

In [ ]:
df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7043 rows x 21 columns]

### Welcome to Day 9 Homework. Please answer the questions below!

1. Buanglah baris dengan `tenure` samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan `tenure` TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

2. Bagi customer yang ***TIDAK MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa ***rata-rata*** dari `tenure` mereka?

3. Bagi customer yang ***MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa maksimal `MonthlyCharges` yang tercata?

4. Berapa banyak customer yang ***TIDAK MEMILIKI PARTNER*** namun ***MEMILIKI DEPENDENTS***?

5. Berapa rata-rata ***TOTAL CHARGES*** dari customer ***NON-SENIOR CITIZEN*** (Senior Citizen = 0) yang memiliki ***PHONE SERVICE*** dan memiliki ***INTERNET SERVICE FIBER OPTIC***?

#### Silakan menuliskan kode untuk menjawab 5 pertanyaan di atas...

In [ ]:
df = df[df['tenure']!=0]
df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7032 rows x 21 columns]

In [ ]:
Non_P_D = df[(df['Partner'] == 'No') & (df['Dependents'] == 'No')].reset_index(drop = True)
Non_P_D

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     5575-GNVDE    Male              0      No         No      34   
1     3668-QPYBK    Male              0      No         No       2   
2     7795-CFOCW    Male              0      No         No      45   
3     9237-HQITU  Female              0      No         No       2   
4     9305-CDSKC  Female              0      No         No       8   
...          ...     ...            ...     ...        ...     ...   
3275  0639-TSIQW  Female              0      No         No      67   
3276  8456-QDAVC    Male              0      No         No      19   
3277  7750-EYXWZ  Female              0      No         No      12   
3278  2569-WGERO  Female              0      No         No      72   
3279  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
0             Yes                No             DSL                  Yes  ...   
1             Yes                No             DSL                  Yes  ...   
2              No  No phone service             DSL                  Yes  ...   
3             Yes                No     Fiber optic                   No  ...   
4             Yes               Yes     Fiber optic                   No  ...   
...           ...               ...             ...                  ...  ...   
3275          Yes               Yes     Fiber optic                  Yes  ...   
3276          Yes                No     Fiber optic                   No  ...   
3277           No  No phone service             DSL                   No  ...   
3278          Yes                No              No  No internet service  ...   
3279          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
0                     Yes                   No                   No   
1                      No                   No                   No   
2                     Yes                  Yes                   No   
3                      No                   No                   No   
4                     Yes                   No                  Yes   
...                   ...                  ...                  ...   
3275                  Yes                   No                  Yes   
3276                   No                   No                  Yes   
3277                  Yes                  Yes                  Yes   
3278  No internet service  No internet service  No internet service   
3279                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
0                      No        One year               No   
1                      No  Month-to-month              Yes   
2                      No        One year               No   
3                      No  Month-to-month              Yes   
4                     Yes  Month-to-month              Yes   
...                   ...             ...              ...   
3275                   No  Month-to-month              Yes   
3276                   No  Month-to-month              Yes   
3277                  Yes        One year               No   
3278  No internet service        Two year              Yes   
3279                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
0                  Mailed check          56.95        1889.5    No  
1                  Mailed check          53.85        108.15   Yes  
2     Bank transfer (automatic)          42.30       1840.75    No  
3              Electronic check          70.70        151.65   Yes  
4              Electronic check          99.65         820.5   Yes  
...                         ...            ...           ...   ...  
3275    Credit card (automatic)         102.95       6886.25   Yes  
3276  Bank transfer (automatic)          7

In [ ]:
mean_NonPD = Non_P_D['tenure'].mean()
mean_NonPD

23.303353658536587

In [ ]:
YesP_NonD = df[(df['Partner'] == 'Yes') & (df['Dependents'] == 'No')].reset_index(drop = True)
YesP_NonD

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     7892-POOKP  Female              0     Yes         No      28   
2     8091-TTVAX    Male              0     Yes         No      58   
3     1680-VDCWW    Male              0     Yes         No      12   
4     3638-WEABW  Female              0     Yes         No      58   
...          ...     ...            ...     ...        ...     ...   
1648  1035-IPQPU  Female              1     Yes         No      63   
1649  7398-LXGYX    Male              0     Yes         No      44   
1650  9281-CEDRU  Female              0     Yes         No      68   
1651  3605-JISKB    Male              1     Yes         No      55   
1652  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
0              No  No phone service             DSL                   No  ...   
1             Yes               Yes     Fiber optic                   No  ...   
2             Yes               Yes     Fiber optic                   No  ...   
3             Yes                No              No  No internet service  ...   
4             Yes               Yes             DSL                   No  ...   
...           ...               ...             ...                  ...  ...   
1648          Yes               Yes     Fiber optic                   No  ...   
1649          Yes               Yes     Fiber optic                  Yes  ...   
1650          Yes                No             DSL                   No  ...   
1651          Yes               Yes             DSL                  Yes  ...   
1652          Yes               Yes     Fiber optic                   No  ...   

         DeviceProtection          TechSupport          StreamingTV  \
0                      No                   No                   No   
1                     Yes                  Yes                  Yes   
2                     Yes                   No                  Yes   
3     No internet service  No internet service  No internet service   
4                      No                  Yes                   No   
...                   ...                  ...                  ...   
1648                  Yes                   No                  Yes   
1649                  Yes                   No                   No   
1650                   No                  Yes                  Yes   
1651                   No                   No                   No   
1652                   No                   No                   No   

          StreamingMovies        Contract PaperlessBilling  \
0                      No  Month-to-month              Yes   
1                     Yes  Month-to-month              Yes   
2                     Yes        One year               No   
3     No internet service        One year               No   
4                      No        Two year              Yes   
...                   ...             ...              ...   
1648                  Yes  Month-to-month              Yes   
1649                   No  Month-to-month              Yes   
1650                   No        Two year               No   
1651                   No        One year               No   
1652                   No  Month-to-month              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
0              Electronic check          29.85         29.85    No  
1              Electronic check         104.80       3046.05   Yes  
2       Credit card (automatic)         100.35        5681.1    No  
3     Bank transfer (automatic)          19.80        202.25    No  
4       Credit card (automatic)          59.90        3505.1    No  
...                         ...            ...           ...   ...  
1648           Electronic check         103.50        6479.4    No  
1649    Credit card (automatic)          8

In [ ]:
YesP_NonD = YesP_NonD[~YesP_NonD['MonthlyCharges'].isnull()].reset_index(drop = True)
YesP_NonD.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1653 entries, 0 to 1652
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        1653 non-null   object 
 1   gender            1653 non-null   object 
 2   SeniorCitizen     1653 non-null   int64  
 3   Partner           1653 non-null   object 
 4   Dependents        1653 non-null   object 
 5   tenure            1653 non-null   int64  
 6   PhoneService      1653 non-null   object 
 7   MultipleLines     1653 non-null   object 
 8   InternetService   1653 non-null   object 
 9   OnlineSecurity    1653 non-null   object 
 10  OnlineBackup      1653 non-null   object 
 11  DeviceProtection  1653 non-null   object 
 12  TechSupport       1653 non-null   object 
 13  StreamingTV       1653 non-null   object 
 14  StreamingMovies   1653 non-null   object 
 15  Contract          1653 non-null   object 
 16  PaperlessBilling  1653 non-null   object 


In [ ]:
max_MC = YesP_NonD['MonthlyCharges'].max()

In [ ]:
max_MC

118.6

In [ ]:
NonP_YesD = df[(df['Partner'] == 'No') & (df['Dependents'] == 'Yes')].reset_index(drop = True)
NonP_YesD

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0    1452-KIOVK    Male              0      No        Yes      22   
1    6388-TABGU    Male              0      No        Yes      62   
2    9959-WOFKT    Male              0      No        Yes      71   
3    8773-HHUOZ  Female              0      No        Yes      17   
4    5630-AHZIL  Female              0      No        Yes       3   
..          ...     ...            ...     ...        ...     ...   
354  9013-AQORL  Female              0      No        Yes      48   
355  6598-KELSS    Male              0      No        Yes      50   
356  5351-QESIO    Male              0      No        Yes       1   
357  6447-GORXK    Male              0      No        Yes      31   
358  7164-BPTUT    Male              0      No        Yes      57   

    PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0            Yes               Yes     Fiber optic             No  ...   
1            Yes                No             DSL            Yes  ...   
2            Yes               Yes     Fiber optic            Yes  ...   
3            Yes                No             DSL             No  ...   
4            Yes                No             DSL            Yes  ...   
..           ...               ...             ...            ...  ...   
354           No  No phone service             DSL             No  ...   
355           No  No phone service             DSL            Yes  ...   
356           No  No phone service             DSL             No  ...   
357           No  No phone service             DSL             No  ...   
358          Yes               Yes             DSL            Yes  ...   

    DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                 No          No         Yes              No  Month-to-month   
1                 No          No          No              No        One year   
2                Yes          No         Yes             Yes        Two year   
3                 No          No         Yes             Yes  Month-to-month   
4                 No         Yes          No             Yes  Month-to-month   
..               ...         ...         ...             ...             ...   
354              Yes          No          No             Yes  Month-to-month   
355               No          No         Yes              No        One year   
356               No          No          No              No  Month-to-month   
357               No          No          No              No  Month-to-month   
358              Yes         Yes         Yes             Yes        Two year   

    PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                Yes    Credit card (automatic)          89.10        1949.4   
1                 No  Bank transfer (automatic)          56.15       3487.95   
2                 No  Bank transfer (automatic)         106.70       7382.25   
3                Yes               Mailed check          64.70        1093.1   
4                Yes  Bank transfer (automatic)          64.50         177.4   
..               ...                        ...            ...           ...   
354               No    Credit card (automatic)          45.55       2108.35   
355              Yes  Bank transfer (automatic)          43.05       2208.05   
356               No               Mailed check          24.20          24.2   
357               No    Credit card (automatic)          25.00         789.2   
358               No               Mailed check          89.55       5012.35   

    Churn  
0      No  
1      No  
2      No  
3     Yes  
4      No  
..    ...  
354    No  
355    No  
356    No  
357    No  
358    No  

[359 rows x 21 columns]

### Silakan menjawab di bawah ini:

#### Soal 1:
Buanglah baris dengan tenure samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan tenure TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

#### Jawaban: 7032 baris

#### Soal 2:
Bagi customer yang TIDAK MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa rata-rata dari tenure mereka?

#### Jawaban: 23.30

#### Soal 3:
Bagi customer yang MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa maksimal MonthlyCharges yang tercatat?

#### Jawaban: 118,6

#### Soal 4:
Berapa banyak customer yang TIDAK MEMILIKI PARTNER namun MEMILIKI DEPENDENTS?

#### Jawaban: 359 customer

#### Soal 5:
Berapa rata-rata TOTAL CHARGES dari customer NON-SENIOR CITIZEN (Senior Citizen = 0) yang memiliki PHONE SERVICE dan memiliki INTERNET SERVICE FIBER OPTIC?

#### Jawaban: 3226.49

In [ ]:
Non_SC = df[(df['SeniorCitizen'] == 0)].reset_index(drop = True)
Non_SC

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
5885  2569-WGERO  Female              0      No         No      72   
5886  6840-RESVB    Male              0     Yes        Yes      24   
5887  2234-XADUH  Female              0     Yes        Yes      72   
5888  4801-JZAZL  Female              0     Yes        Yes      11   
5889  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
0              No  No phone service             DSL                   No  ...   
1             Yes                No             DSL                  Yes  ...   
2             Yes                No             DSL                  Yes  ...   
3              No  No phone service             DSL                  Yes  ...   
4             Yes                No     Fiber optic                   No  ...   
...           ...               ...             ...                  ...  ...   
5885          Yes                No              No  No internet service  ...   
5886          Yes               Yes             DSL                  Yes  ...   
5887          Yes               Yes     Fiber optic                   No  ...   
5888           No  No phone service             DSL                  Yes  ...   
5889          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
0                      No                   No                   No   
1                     Yes                   No                   No   
2                      No                   No                   No   
3                     Yes                  Yes                   No   
4                      No                   No                   No   
...                   ...                  ...                  ...   
5885  No internet service  No internet service  No internet service   
5886                  Yes                  Yes                  Yes   
5887                  Yes                   No                  Yes   
5888                   No                   No                   No   
5889                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
0                      No  Month-to-month              Yes   
1                      No        One year               No   
2                      No  Month-to-month              Yes   
3                      No        One year               No   
4                      No  Month-to-month              Yes   
...                   ...             ...              ...   
5885  No internet service        Two year              Yes   
5886                  Yes        One year              Yes   
5887                  Yes        One year              Yes   
5888                   No  Month-to-month              Yes   
5889                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
0              Electronic check          29.85         29.85    No  
1                  Mailed check          56.95        1889.5    No  
2                  Mailed check          53.85        108.15   Yes  
3     Bank transfer (automatic)          42.30       1840.75    No  
4              Electronic check          70.70        151.65   Yes  
...                         ...            ...           ...   ...  
5885  Bank transfer (automatic)          21.15        1419.4    No  
5886               Mailed check          8

In [ ]:
NonSC_PI = Non_SC[(Non_SC['PhoneService'] == 'Yes') & (Non_SC['InternetService'] == 'Fiber optic')].reset_index(drop = True)
NonSC_PI

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     9237-HQITU  Female              0      No         No       2   
1     9305-CDSKC  Female              0      No         No       8   
2     1452-KIOVK    Male              0      No        Yes      22   
3     7892-POOKP  Female              0     Yes         No      28   
4     8091-TTVAX    Male              0     Yes         No      58   
...          ...     ...            ...     ...        ...     ...   
2260  9767-FFLEM    Male              0      No         No      38   
2261  0639-TSIQW  Female              0      No         No      67   
2262  8456-QDAVC    Male              0      No         No      19   
2263  2234-XADUH  Female              0     Yes        Yes      72   
2264  3186-AJIEK    Male              0      No         No      66   

     PhoneService MultipleLines InternetService OnlineSecurity  ...  \
0             Yes            No     Fiber optic             No  ...   
1             Yes           Yes     Fiber optic             No  ...   
2             Yes           Yes     Fiber optic             No  ...   
3             Yes           Yes     Fiber optic             No  ...   
4             Yes           Yes     Fiber optic             No  ...   
...           ...           ...             ...            ...  ...   
2260          Yes            No     Fiber optic             No  ...   
2261          Yes           Yes     Fiber optic            Yes  ...   
2262          Yes            No     Fiber optic             No  ...   
2263          Yes           Yes     Fiber optic             No  ...   
2264          Yes            No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No         Yes             Yes  Month-to-month   
2                  No          No         Yes              No  Month-to-month   
3                 Yes         Yes         Yes             Yes  Month-to-month   
4                 Yes          No         Yes             Yes        One year   
...               ...         ...         ...             ...             ...   
2260               No          No          No              No  Month-to-month   
2261              Yes          No         Yes              No  Month-to-month   
2262               No          No         Yes              No  Month-to-month   
2263              Yes          No         Yes             Yes        One year   
2264              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          70.70        151.65   
1                 Yes           Electronic check          99.65         820.5   
2                 Yes    Credit card (automatic)          89.10        1949.4   
3                 Yes           Electronic check         104.80       3046.05   
4                  No    Credit card (automatic)         100.35        5681.1   
...               ...                        ...            ...           ...   
2260              Yes    Credit card (automatic)          69.50       2625.25   
2261              Yes    Credit card (automatic)         102.95       6886.25   
2262              Yes  Bank transfer (automatic)          78.70        1495.1   
2263              Yes    Credit card (automatic)         103.20        7362.9   
2264              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0      Yes  
1      Yes  
2       No  
3      Yes  
4       No  
...    ...  
2260    No  
2261   Yes  
2262    No  
2263    No  
2264    No  

[2265 rows x 21 columns]

In [ ]:
NonSC_PI[NonSC_PI['TotalCharges'].isnull()]

Empty DataFrame
Columns: [customerID, gender, SeniorCitizen, Partner, Dependents, tenure, PhoneService, MultipleLines, InternetService, OnlineSecurity, OnlineBackup, DeviceProtection, TechSupport, StreamingTV, StreamingMovies, Contract, PaperlessBilling, PaymentMethod, MonthlyCharges, TotalCharges, Churn]
Index: []

[0 rows x 21 columns]

In [ ]:
NonSC_PI = NonSC_PI.astype({'TotalCharges':'float'})

In [ ]:
NonSC_PI.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2265 entries, 0 to 2264
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        2265 non-null   object 
 1   gender            2265 non-null   object 
 2   SeniorCitizen     2265 non-null   int64  
 3   Partner           2265 non-null   object 
 4   Dependents        2265 non-null   object 
 5   tenure            2265 non-null   int64  
 6   PhoneService      2265 non-null   object 
 7   MultipleLines     2265 non-null   object 
 8   InternetService   2265 non-null   object 
 9   OnlineSecurity    2265 non-null   object 
 10  OnlineBackup      2265 non-null   object 
 11  DeviceProtection  2265 non-null   object 
 12  TechSupport       2265 non-null   object 
 13  StreamingTV       2265 non-null   object 
 14  StreamingMovies   2265 non-null   object 
 15  Contract          2265 non-null   object 
 16  PaperlessBilling  2265 non-null   object 


In [ ]:
mean_SCPI = NonSC_PI['TotalCharges'].mean()
mean_SCPI

3226.4927373068435